### ISSUE TITLE
Resize Tool Mostly Broken

### ISSUE THREAD

**Issue by HuntJSparra** [bokeh/bokeh#4560]
Following Issues I have had with resize tool:

- 0.11.1, possibly earlier: Resizing an axis to a near zero values will break the plot (resetting does not help, only reloading the page does), immediately attempting to resize another plot will break it as well (in dev12 all plots are broken); Error: Unkown Constraint

- 0.12.0dev12: Resizing plot blurs it across the original area

- 0.12.0dev12: Tools can still be used in the initial area even if the plot has been scaled to a smaller area

- 0.12.0dev12: Scaling to a larger size clips everything outside the initial plot area

- 0.12.0dev12, possibly earlier dev builds: Resizing on a grid plot does not shift other plots

- 0.12dev12, possibly earlier dev builds: Resetting after resizing a plot snaps the toolbar to the resized area instead of the reset area; resetting again fixes this

<img width="505" alt="screen shot 2016-06-17 at 1 35 00 pm" src="https://cloud.githubusercontent.com/assets/19802403/16160910/664be462-3490-11e6-9782-7cad081f7f6a.png">

**Comment by birdsarah** [comment]
I wouldn't characterize it as broken. 

I think you may be testing it in "unfair" circumstances. e.g. in a grid plot - the purpose of which is to have a series of equally sized plots - the problem is that the grid plot doesn't update, not the resize tool.

Can you elaborate all of these uses more - what you tried, what you expected, what you saw. Then we can evaluate whether we want to support each use case.

**Comment by bryevdv** [comment]
@birdsarah I can confirm in current master the resize tool is basically completely broken. It leaves behind partially unrendered junk, overruns toolbars, etc. I think we should just get rid of it (print a deprecation, not break anything)

**Comment by HuntJSparra** [comment]
In the same order:

- Expected after resizing to a near-zero value I would be able to resize it back to normal, but was not able to. imo, there is no practical use for resizing it so small but is still an issue if it forces me to reload the entire page. Occurs with every plot with a resize tool currently.

- Expected old visuals to be rendered over. Occurs with every plot with a resize tool currently, although I would not be surprised if next dev build fixed it. EDIT: Have not tested on dev13

- After resizing I expected to not be able to use the pan tool outside the now smaller plot, but was able to. Tried also with the box zoom. Occurs with every plot with a resize tool currently, although I would not be surprised if next dev build fixed it. EDIT: Have not tested on dev13

- I attempted to resize my plot to be larger, but just found that all parts outside the original plot were just clipped. Occurs with every plot with a resize tool currently, although I would not be surprised if next dev build fixed it. EDIT: Have not tested on dev13

- If its the grid plot not updating then I really do not have anything to add. Just expected it to shift other plots around like in 0.11

**Comment by almarklein** [comment]
I can confirm this behavior. It appears that the plot is resized, but the actual canvas is not (and leaving behind "ghosts" of the plots). There is at least one example (`plotting/file/categorical.py`) that has the resize tool on by default, which leads to really odd behavior when trying to interact with that plot.

**Comment by almarklein** [comment]
(occurs anywhere where the resize tool is available, this behavior does not seem specific to being in a certain layout).

**Comment by rodrigotp** [comment]
I have updated to Bokeh 0.12 and I can confirm the resize tool is broken. I will now revert back to version 0.11. 

For me, the resizing tool is one of the best tools in Bokeh, and it is a pitty if it will be discontinued. I use it to print PSCAD data, and when we need to debug a signal in our model, it is very handy to be able to increase that graphic size and see magnified what is going on. 

It will be a pitty if this feature is abandoned.

**Comment by bryevdv** [comment]
> it is very handy to be able to increase that graphic size and see magnified what is going on.

Just curious, another tool that's been suggested is a magnifier/spyglass tool. Basically you hover over a point on a plot, and a magnified image of the area under the cursor shows up in hover or maybe a separate div. That seems like it might actually help your described use case even better? FWIW that sort of thing could be implemented today as a custom user extension, tho it seems useful enough to be considered for the core library too.

**Comment by bryevdv** [comment]
OK @pzwang has outlines some other reasonable use cases of resizing, if not for individual plots, then maybe for each layout root. @birdsarah let's try to discuss more when you return

**Comment by rodrigotp** [comment]
I think the magnifier is also a good idea, and I believe it would work even better in conjunction with the resize tool, but it is not a substitute. Sometimes we want to better analyse a group of subplots (e.g. 12 of them in a same page) and guide that analysis based on one that we actually resized to take a better look. Then I guess the magnifying glass will be useful to look into the remaining 11 subplots. But if the resize is not available, in the one subplot I want to make bigger, once the magnifier glass is moved, we would lose the expanded view of the one subplot we used to drive the analysis. Does it make sense?

Another argument is the awe factor. Until now it has really astonished the people to which I have shown it.
